lightning/src/chain/channelmonitor.rs

```diff
@@ -1425,39 +1507,61 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 				check_htlc_fails!(txid, "previous", 'prev_loop);
 			}
 
+			let (htlc_claim_reqs, set_script) = self.get_counterparty_htlc_output_claim_reqs(commitment_number, commitment_txid, Some(tx));
+			if set_script {
+				self.counterparty_payment_script = {
```

Comment:
I'm somewhat confused here - we always set this in the constructor and the re-setting code here is identical to that, why do it again (I know there's another place where we do it - but I'm not sure if can't just skip it in both places).

Reply:
Digging into the `git blame`, it seems like we first started setting `self.counterparty_payment_script` (formerly named `remote_payment_script`) outside of the constructor here: https://github.com/rust-bitcoin/rust-lightning/commit/0a345c046caeb6cd5af1b1c3dfc093b450cb6cee#diff-3f5233d06119105f41269504316d23871aeef9bfa72f7a63782f2bf3d193f135R1570 

But I'm unclear why we started setting it there?? It seems purposeful in that commit but I can't tell why, so I'm inclined to remove it.

Reply:
I guess that's a leftover from pre-`option_static_remotekey`, where countersignatory's P2WPKH pubkey was computed with the `per_commitment_point`, thus conditional on the state broadcasted. It can goes.